test/integration/evictions/evictions_test.go

```diff
@@ -184,30 +188,31 @@ func TestConcurrentEvictionRequests(t *testing.T) {
 				}
 			}
 			if e != nil {
-				if errors.IsConflict(err) {
-					fmt.Errorf("Unexpected Conflict (409) error caused by failing to handle concurrent PDB updates: %v", e)
-				} else {
-					errCh <- e
+				if errors.IsConflict(e) {
+					e = fmt.Errorf("Unexpected Conflict (409) error caused by failing to handle concurrent PDB updates: %v", e)
 				}
+				errCh <- e
```

Comment:
Is returning from the goroutine after receiving a conflict error the correct thing to do? nothing will be sent over `doneCh` in that case and wouldn't we fall through to the timeout case on line 213? Five minutes might be a long time to wait before moving on with the rest of the test. Have you collected any rough estimates on how long you expect evicting 10 pods to take when there isn't an error? I am aware that using channels might be the more "canonical" design pattern to communicate the status of the goroutine  (e.g. error encountered/no problems) but it might be simpler to use a `sync.WaitGroup` and make each goroutine responsible for quitting after the default timeout. A buffered `errCh` could allow you to store errors concurrently and process them later.